packages/lib/serverConfig.ts

```diff
@@ -13,10 +13,12 @@ function detectTransport(): SendmailTransport.Options | SMTPConnection.Options |
     const transport = {
       host: process.env.EMAIL_SERVER_HOST,
       port,
-      auth: {
-        user: process.env.EMAIL_SERVER_USER,
-        pass: process.env.EMAIL_SERVER_PASSWORD,
-      },
+      auth: process.env.EMAIL_SERVER_PASSWORD
```

Comment:
I think a better approach here would be to use `isENVProd` check instead of the server password. Would make the use case more clearer

Reply:
what if the user tries to use google smtp server on the local machine then this won't work with isENVProd right?

Reply:
I think that is happening here is that mailhog requires no authentication, and setting auth: to `process.env.EMAIL_SERVER_PASSWORD` disables auth implicitly (as this is probably an empty string).

instead of this, I'd suggest removing the "auth" key entirely as per https://stackoverflow.com/a/73117417 when the pass/user are empty